app/stac_api/views.py

```diff
@@ -481,6 +483,24 @@ def get_serializer(self, *args, **kwargs):
         kwargs.setdefault('context', self.get_serializer_context())
         return serializer_class(*args, **kwargs)
 
+    def perform_upsert(self, serializer, lookup):
+        if serializer.validated_data['name'] != self.kwargs['asset_name']:
+            raise ValidationError(
+                _('This query leads to a rename of the asset'
+                  ' and this is forbidden (in case you want to rename,'
+                  ' you have to delete the asset first).')
```

Comment:
actually with the presigned url it will be 'creating new' and then 'deleting old'

Reply:
This code could be actually be moved inside a generic function for all renaming:
```python
# validators_view.py
def validate_renaming(serializer, original_name, log_extra):
  if serializer.validated_data['name'] !=original_name:
    logger.error(..., extra=log_extra)
     raise ValidationError(_('Renaming object is not allowed'), code='invalid')
```
This should also then be used fro all renaming like for the collection (@IsabelleBzr)

@rebert  note also that in your implementation logging has been forgotten.